cpp/tests/types/types_test.cpp

```diff
@@ -292,6 +318,16 @@ TEST_F(WrappersTestBool8, CastArithmeticTest)
               static_cast<bool>(static_cast<int>(t1) + static_cast<int>(t2)));
 }
 
+// Ensure bool8 constructor is correctly casting the input type to a bool
+TYPED_TEST(Bool8CtorTest, Bool8CastTest) {
+using SourceType = TypeParam;
+  for (int i = 0; i < NUM_TRIALS; ++i) {
+    SourceType t{this->rand()};
+    cudf::bool8 const w{t};
+    EXPECT_EQ(unwrap(w), static_cast<bool>(t));
```

Comment:
I'm not sure this is testing what we want. This will just verify that `unwrap(w)` and  `static_cast<bool>(t)` are greater than zero.

I _think_ what we want to test is this:
```
    EXPECT_EQ(unwrap(w), static_cast<bool8::value_type>(static_cast<bool>(t))));
```

This will ensure that the underlying value of `w` is identical to the value of `t`.